kafkajobs/logger.go

```diff
@@ -53,3 +51,40 @@ func (l *logger) Log(level kgo.LogLevel, msg string, keyvals ...any) {
 		l.l.Debug(msg, zf...)
 	}
 }
+
+func toKeyValuePair(keyvals []any) []zapcore.Field {
+	inLen := len(keyvals)
+	if inLen == 0 {
+		return nil
+	}
+
+	if inLen%2 == 0 {
+		// Make sure we have even number of elements and every odd element is a string.
+		isKVP := true
+		keys := make([]string, 0, inLen/2)
+		for i := 0; i < inLen; i += 2 {
+			if key, ok := keyvals[i].(string); ok {
+				keys = append(keys, key)
+			} else {
+				isKVP = false
+				break
+			}
+		}
+
+		// positive scenario
+		if isKVP {
+			result := make([]zap.Field, inLen/2)
+			for i, key := range keys {
+				result[i] = zap.Any(key, keyvals[i*2+1])
+			}
+			return result
+		}
```

Comment:
This is the hot place, don't iterate twice here (keys check + keys add to the result). 
Remove `isKVP`, because if by some reason the number is odd, as I suggested, just add some placeholder at the end of the sequence with some predefined text.
Preallocate `result` slice and while iterating - add key-value pairs to the result.
Do not cast values to `string` type, it does not matter since you're using `zap.Any`.